### ISSUE TITLE
Implement TCP_USER_TIMEOUT in HttpClientOptions

### ISSUE THREAD

**Issue by rbsrafa** [eclipse-vertx/vert.x#4427]
### Version

Which version(s) did you encounter this bug ? 
4.3.2-SNAPSHOT (Master) and 3.9.9 (release)

### Context

When using Vertx HttpClient with HTTP2 configured, the client gets stuck in an established TCP 
connection in case the FIN (finish) and RST (reset) packets are not transmitted/lost
from the server. 

### Do you have a reproducer?

Please have a look on this [Github Repo](https://github.com/rbsrafa/test-vertx) README.md to follow the steps to reproduce it.

### Extra

Running on:
```shell
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Comment by rbsrafa** [comment]
I've created a pull request for this issue: https://github.com/eclipse-vertx/vert.x/pull/4428/files

Please let me know if I missed something. I also would like to know if this could be back-ported to the version 3.9.9. Thanks.

**Comment by tsegismont** [comment]
@rbsrafa when #4428 is merged, I'll take care of backporting.

**Comment by tsegismont** [comment]
Closed in #4428
